private/buf/bufmigrate/migrator.go

```diff
@@ -456,6 +457,10 @@ func (m *migrator) getModuleToRefToCommit(
 	ctx context.Context,
 	moduleRefs []bufmodule.ModuleRef,
 ) (map[string]map[string]bufmodule.Commit, error) {
+	// We need to dedupe all the module refs of the configured dependencies, since in `v1`
+	// workspaces, each module will have their own configured dependencies, which may be
+	// duplicated across modules.
+	moduleRefs = dedupeStringerSlice(moduleRefs)
```

Comment:
What if I have `buf.build/foo/bar:label1` and `buf.build/foo/bar:label2`, what do you want the behavior to be?

Reply:
We have logic that follows up this that takes the latest version if there are multiple pins: https://github.com/bufbuild/buf/blob/f4d2a422f9d64cc4820200457fa07fb1d48ce275/private/buf/bufmigrate/migrator.go#L559

The dedupe logic is purely so that we are not resolving the same module key and and module ref multiple times, since this causes issues with our providers.

Reply:
Let's chat about this live - I'm inclined to just merge since it's just migrate, but (1) providers shouldn't be wonky if we provide duplicates (2) want to make sure this logic is clear to others, optimally

Reply:
Chatted live, just commenting for posterity: added context around why `bufmodule.ModuleKeyProvider` and `bufmodule.CommitProvider` expects implementations to error for duplicate keys (avoiding unnecessary expensive operations like API calls to resolve these entities).